javav2/example_code/scheduler/src/main/java/com/example/eventbrideschedule/scenario/EventbridgeSchedulerScenario.java

```diff
@@ -0,0 +1,365 @@
+     * If any errors occur during the cleanup process, the corresponding error messages are logged.
+     */
+    public static void cleanUp() {
+        logger.info("First, delete the schedule group.");
+        waitForInputToContinue();
+        try {
+            eventbridgeActions.deleteScheduleGroupAsync(scheduleGroupName).join();
+
+        } catch (CompletionException ce) {
+            Throwable cause = ce.getCause();
+            if (cause instanceof SchedulerException schedulerException) {
+                logger.error("Scheduler error occurred: Error message: {}, Error code {}",
+                    schedulerException.getMessage(), schedulerException.awsErrorDetails().errorCode(), schedulerException);
+            } else {
+                logger.error("An unexpected error occurred: {}", cause.getMessage());
+            }
+            return;
+        }
+        logger.info("Next, delete the schedules");
+        waitForInputToContinue();
+        try {
+            eventbridgeActions.deleteScheduleAsync(recurringScheduleName, scheduleGroupName).join();
+            eventbridgeActions.deleteScheduleAsync(oneTimeScheduleName, scheduleGroupName).join();
+
+        } catch (CompletionException ce) {
+            Throwable cause = ce.getCause();
+            if (cause instanceof SchedulerException schedulerException) {
+                logger.error("Scheduler error occurred: Error message: {}, Error code {}",
+                    schedulerException.getMessage(), schedulerException.awsErrorDetails().errorCode(), schedulerException);
+            } else {
+                logger.error("An unexpected error occurred: {}", cause.getMessage());
+            }
+            return;
+        }
+
+        logger.info("Finally, destroy the CloudFormation stack");
+        waitForInputToContinue();
+        CloudFormationHelper.destroyCloudFormationStack(STACK_NAME);
+    }
+
+    /**
+     * Prepares the application by creating resources in a CloudFormation stack, including an SNS topic
+     * that will be subscribed to the EventBridge Scheduler events. The user will need to confirm the subscription
+     * in order to receive event emails.
+     *
+     * @return true if the application preparation was successful, false otherwise
+     */
+    public static boolean prepareApplication() {
+        logger.info("""
+            This example creates resources in a CloudFormation stack, including an SNS topic
+            that will be subscribed to the EventBridge Scheduler events.
+            You will need to confirm the subscription in order to receive event emails.
+             """);
+
+        String emailAddress = promptUserForEmail();
+        logger.info("You entered {}", emailAddress);
+
+        logger.info("Get the roleArn and snsTopicArn values using a Cloudformation template.");
+        waitForInputToContinue();
+        CloudFormationHelper.deployCloudFormationStack(STACK_NAME, emailAddress);
```

Comment:
The spec says get the stack name from the user. 
I don't know if this matters. Just letting you know.

Reply:
I now ask user if they want to use a new stack name.